src/common/event-bus.ts

```diff
@@ -0,0 +1,51 @@
+export interface Listener<T> {
+  (event: T): void;
+}
+
+export interface Disposable {
+  dispose(): void;
+}
+
+export class EventBus<T> {
```

Comment:
Is it differs somehow from standard node's event-emitter?
In addition we have already one more custom event-emitter in `renderer/utils/eventEmitter.ts` (could be moved to `common/utils`)

Reply:
I didn't notice that we already have EventEmitter. No need for yet-another -> I'll refactor this a bit.

Reply:
Refactored.